source/1_engine/rex_engine/include/rex_engine/primitives/box.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "rex_engine/primitives/mesh_factory.h"
+
+namespace rex
+{
+    namespace mesh_factory
+    {
+        template<typename T>
+        MeshData<T> create_box(f32 width, f32 height, f32 depth, u32 subdivisions)

```

Comment:
This might not need to be templated but other data structures might require this.
The idea of this template was to lower the resolution not increment it ( eg: s16 ).